packages/@aws-cdk/integ-tests-alpha/lib/assertions/sdk.ts

```diff
@@ -250,6 +250,19 @@ export class LambdaInvokeFunction extends AwsApiCall {
       arnFormat: ArnFormat.COLON_RESOURCE_NAME,
       resourceName: props.functionName,
     })]);
+
+    // If using `waitForAssertions`, do the same for `waiterProvider` as above.
+    Aspects.of(this).add({
```

Comment:
Although we have created a Resource-based Policy for Lambda in this part, I have not created it here because it is possible to invoke Lambda from Lambda if the appropriate Policy is granted to the invoking Lambda's execution role.
https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/integ-tests-alpha/lib/assertions/sdk.ts#L236

If you have any concerns, I would really appreciate it if you could let me know.

Reply:
I'm fine with this but -- can you document why you've used an Aspect so that others who stumble on your code will understand the decision please